docs/src/modules/spring/pages/views.adoc

```diff
@@ -1 +1,186 @@
+
+
+[#topic-view]
+== Creating a View from a topic
+
+The source of a View can be a topic. It works the same way as shown in <<event-sourced-entity>> or <<value-entity>>, but you define it with `@Subscribe.Topic` instead.
+
+
+[#transform-results]
+== How to transform results
+
+When creating a View, you can transform the results as a relational projection instead of using a `SELECT *` statement.
+
+ifdef::review[REVIEWERS: it would be nice to have use cases describing why they might want to use these different transformation techniques. And does this information apply to all views, regardless of whether they were created from entities or topics?]
+
+=== Relational projection
+
+Instead of using `SELECT *` you can define what columns that will be used in the response message:
+
+
+In a similar way, you can include values from the request message in the response, for example `:request_id`:
+
+[source,proto,indent=0]
+----
+SELECT :request_id, customer_id as id, name FROM customers WHERE name = :customer_name
+----
+
+=== Response message including the result
+
+Instead of streamed results you can include the results in a repeated field in the response message:
+
+
+
+
+[#changing]
+== How to modify a View
+
+Kalix creates indexes for the View based on the query. For example, the following query will result in a View with an index on the `name` column:
+
+[source,proto,indent=0]
+----
+SELECT * FROM customers WHERE name = :customer_name
+----
+
+If the query is changed, Kalix might need to add other indexes. For example, changing the above query to filter on the `city` would mean that Kalix needs to build a View with the index on the `city` column.
+
+[source,proto,indent=0]
+----
+SELECT * FROM customers WHERE address.city = :city
+----
+
+Such changes require you to define a new View. Kalix will then rebuild it from the source event log or value changes.
+
+WARNING: Views from topics cannot be rebuilt from the source messages, because it's not possible to consume all events from the topic again. The new View will be built from new messages published to the topic.
+
+Rebuilding a new View may take some time if there are many events that have to be processed. The recommended way when changing a View is multi-step, with two deployments:
+
+
+. Define the new View, and keep the old View intact. A new View is defined by a new View??. The `viewId` is the same as the `service` name, i.e. it will be a different `viewId` than the old View. Keep the old `register` of the old `service` in `Main`.
```

Comment:
> A new View is defined by a new View??

Still draft PR? 

Maybe good to add very clear markers otherwise we risk to miss this kind of thing and end up merging it.

Reply:
We can remove that sentence actually. I mean "A new View is defined by a new View". 

Users should define a new View and keep the old, that's All. The service name is the name of the class.

Reply:
Which makes me realise now that users should be free to change the name of their class without breaking the view. If they do that today, they will create a new view and rebuild it from scratch. They may not even notice it.

Reply:
> Keep the old `register` of the old `service` in `Main`

Not applicable

Reply:
You are right. It should have been a Draft PR.
removed both. 

> Which makes me realise now that users should be free to change the name of their class without breaking the view. If they do that today, they will create a new view and rebuild it from scratch. They may not even notice it.

Is there anything we should add to this regard?

Reply:
make sure all our view include a viewId . We highly recommend to have an ViewId

Reply:
create an issue about nott clear transformations in view in protocol first.

Reply:
As discussed on call, my memory failed me. We do have already a `ViewId` and if not defined we use the class name. Not great. We should make `ViewId` mandatory. I will create a ticket for it.

That said, the paragraph that explain how to create a new view by change the view id should only be re-written to explain the annotation.

@franciscolopezsancho, can you include a paragraph explaining that the usage of `ViewId` is highly recommended? 

We should also have it on each View that have in the samples, but can be on another PR.

Reply:
roger that